### ISSUE TITLE
Invoking (+1 x) at the REPL confuses the reader.

### ISSUE THREAD

**Issue by drmeister** [drmeister/clasp#32]
This should cause an "illegal function call error" but the reader is getting confused.

**Comment by Shinmera** [comment]
Can confirm on b05884f.
```
> (+1 4)
terminate called after throwing an instance of 'std::invalid_argument'
  what():  mpz_set_str
src/gctools/memoryManagement.cc:72 Trapped SIGABRT - starting debugger
ABORT was called!!!!!!!!!!!!

"# 61/b@0x7f79fbbfa3d1         readtable.cc  1236/0   (CORE:READER-LIST-ALLOW-CONSING-DOT #<COMMON-LISP:SYNONYM-STREAM @0x7f7a0d223970) >  #\\( )
" 
"The following restarts are available:" 
"ABORT      a    Abort to REPL" 
Frame-61-Dbg[2]>
```

**Comment by drmeister** [comment]
Fixed in 5e75c9d

**Comment by Shinmera** [comment]
Not so fast. On 9c24f9a45d0bd4bd4a80cdcac18b7cee012947fb

```
> '(+1 2)   

(1 2)
> '+1.

1
> '+1.0

1.0
> +1/2
terminate called after throwing an instance of 'std::invalid_argument'
  what():  mpz_set_str
../../src/gctools/memoryManagement.cc:88 Trapped SIGABRT - starting debugger
ABORT was called!!!!!!!!!!!!
```

**Comment by drmeister** [comment]
Fixed in b79c19c

**Comment by Shinmera** [comment]
Fixed on 3b83c17c666a05ab4f23aac5d13335edff463aec
